numpy/lib/polynomial.py

```diff
@@ -678,7 +678,10 @@ def polyval(p, x):
         x = NX.asarray(x)
         y = NX.zeros_like(x)
```

Comment:
One way to get it right is to change this to
```
y = (NX.ones_like(x) * p[0]) if len(p) > 0 else NX_zeros_like(x)
```
and then run over `range(1, len(p))` below.
(Obviously, l. 676 should be adjusted as well.)